apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+      <div className={styles.soundEntryMain}>
+        <div
+          className={classNames(
+            styles.iconContainer,
+            SoundStyle[sound.type]?.classNameBackground,
+            SoundStyle[sound.type]?.classNameBorder,
+            isSelected && styles.iconContainerSelected
+          )}
+        >
+          <FontAwesomeV6Icon
+            iconName={SoundStyle[sound.type]?.icon || ''}
+            className={classNames(
+              styles.typeIcon,
+              SoundStyle[sound.type]?.classNameColor
+            )}
+          />
+        </div>
+        <div
+          className={classNames(
+            styles.name,
+            sound.type === 'vocal' && styles.nameVocal
+          )}
+        >
+          {sound.name}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+// ScrollIntoView: This component does an eased scroll of the container's content,
+// starting from a distance below the top, and scrolling to the top.  It's useful
+// to show the user that some new content is scrollable.
+
+interface ScrollIntoViewProps {
+  id?: string;
+  className?: string;
+  doScroll: boolean;
+  delay: number;
+  frames: number;
+  distanceY: number;
+  children: React.ReactNode;
+}
+
+const ScrollIntoView: React.FunctionComponent<ScrollIntoViewProps> = ({
+  id,
+  className,
+  doScroll,
+  delay,
+  frames,
+  distanceY,
+  children,
+}) => {
+  const scrollStep = useRef<number | undefined>(0);
+  const lastScrollPosition = useRef<number | undefined>(undefined);
+  const containerRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+
+  const containerRefCallback = (ref: HTMLDivElement) => {
+    containerRef.current = ref;
+  };
+
+  function easeOutSine(x: number): number {
+    return Math.sin((x * Math.PI) / 2);
+  }
+
+  // Initial render.
+  useEffect(() => {
+    if (!doScroll) {
+      return;
+    }
+
+    setInterval(() => {
+      const manualScrollDetectionThreshold = 5;
+
+      if (scrollStep.current !== undefined && containerRef.current) {
+        if (
+          lastScrollPosition.current &&
+          Math.abs(
+            containerRef.current?.scrollTop - lastScrollPosition.current
+          ) > manualScrollDetectionThreshold
+        ) {
+          // The user appears to have scrolled manually, so stop doing the
+          // automatic scroll now.
+          scrollStep.current = undefined;
+        } else {
+          // Update the automatic scroll.
+          const progress = Math.max(0, (scrollStep.current - delay) / frames);
+          const scrollPosition = distanceY - distanceY * easeOutSine(progress);
+
+          containerRef.current?.scroll(0, scrollPosition);
+
+          lastScrollPosition.current = scrollPosition;
+
+          scrollStep.current++;
+
+          if (scrollStep.current > delay + frames) {
+            // The automatic scroll has reached its destination.
+            scrollStep.current = undefined;
+          }
+        }
+      }
+    }, 1000 / 60);
```

Comment:
Should this interval rate also be configurable via props?

Reply:
I think 60fps is reasonable for the component to own, but will make a constant.